src/LazyDefinition.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Definitions;
+
+use ProxyManager\Factory\LazyLoadingValueHolderFactory;
+use Psr\Container\ContainerInterface;
+use Yiisoft\Definitions\Contract\DefinitionInterface;
+use Yiisoft\Definitions\Helpers\Normalizer;
+
+final class LazyDefinition implements DefinitionInterface
+{
+    public function __construct(
+        private mixed $definition,
+        /**
+         * @var class-string
+         */
+        private string $objectClass,
+    ) {
+    }
+
+    /**
+     * @psalm-suppress MixedArgumentTypeCoercion
+     */
+    public function resolve(ContainerInterface $container): mixed
```

Comment:
```suggestion
    public function resolve(ContainerInterface $container): VirtualProxyInterface
```

And  won't need ` @psalm-suppress MixedArgumentTypeCoercion`